jax_distribution/systems/rec_mappo_rware.py

```diff
@@ -409,17 +382,14 @@ def _loss_fn(
             shuffled_batch = jax.tree_util.tree_map(
                 lambda x: jnp.take(x, permutation, axis=1), batch
             )
-            minibatches = jax.tree_util.tree_map(
-                lambda x: jnp.swapaxes(
-                    jnp.reshape(
-                        x,
-                        [x.shape[0], config["num_minibatches"], -1] + list(x.shape[2:]),
-                    ),
-                    1,
-                    0,
+            reshaped_batch = jax.tree_util.tree_map(
+                lambda x: jnp.reshape(
+                    x,
+                    [x.shape[0], config["num_minibatches"], -1] + list(x.shape[2:]),
```

Comment:
Can we change this slightly, here and in the other PR::

```suggestion
                    x, (x.shape[0], config["num_minibatches"], -1, *x.shape[2:])
```